apps/SageAccountWeb/src/hooks/useMaybeRedirectToSignTermsOfService.test.ts

```diff
@@ -0,0 +1,204 @@
+    })
+  })
+  it('redirects to signTermsOfUse when user ToS is MUST_AGREE_SOON and user has not skipped signing updated ToS and has never signed ToU', async () => {
+    sessionStorage.removeItem(SKIPPED_SIGNING_TOS_SESSIONSTORAGE_KEY)
+    mockUseApplicationSessionContext.mockReturnValue({
+      ...mockApplicationSessionContext,
+      termsOfServiceStatus: {
+        userId: '1234',
+        userCurrentTermsOfServiceState: TermsOfServiceState.MUST_AGREE_SOON,
+        lastAgreementDate: null,
+      },
+    })
+
+    const hook = renderHook(() => useMaybeRedirectToSignTermsOfService())
+    await waitFor(() => {
+      expect(hook.result.current.mayPromptTermsOfUse).toBe(true)
+      expect(mockNavigate).toHaveBeenCalledWith('/authenticated/signTermsOfUse')
+    })
+  })
+  it('redirects to signTermsOfUse when user ToS is MUST_AGREE_NOW and has never signed ToU', async () => {
+    mockUseApplicationSessionContext.mockReturnValue({
+      ...mockApplicationSessionContext,
+      termsOfServiceStatus: {
+        userId: '1234',
+        userCurrentTermsOfServiceState: TermsOfServiceState.MUST_AGREE_NOW,
+        lastAgreementDate: null,
+      },
+    })
+
+    const hook = renderHook(() => useMaybeRedirectToSignTermsOfService())
+    await waitFor(() => {
+      expect(hook.result.current.mayPromptTermsOfUse).toBe(true)
+      expect(mockNavigate).toHaveBeenCalledWith('/authenticated/signTermsOfUse')
+    })
+  })
+
+  it('redirects to signUpdatedTermsOfUse when user ToS is MUST_AGREE_SOON and user has not skipped signing updated ToS and has signed ToU', async () => {
+    sessionStorage.removeItem(SKIPPED_SIGNING_TOS_SESSIONSTORAGE_KEY)
+    mockUseApplicationSessionContext.mockReturnValue({
+      ...mockApplicationSessionContext,
+      termsOfServiceStatus: {
+        userId: '1234',
+        userCurrentTermsOfServiceState: TermsOfServiceState.MUST_AGREE_SOON,
+        lastAgreementDate: new Date().toISOString(),
+      },
+    })
+
+    const hook = renderHook(() => useMaybeRedirectToSignTermsOfService())
+    await waitFor(() => {
+      expect(hook.result.current.mayPromptTermsOfUse).toBe(true)
+      expect(mockNavigate).toHaveBeenCalledWith(
+        '/authenticated/signUpdatedTermsOfUse',
+      )
+    })
+  })
+  it('redirects to signTermsOfUse when user ToS is MUST_AGREE_NOW and has signed ToU', async () => {
+    mockUseApplicationSessionContext.mockReturnValue({
+      ...mockApplicationSessionContext,
+      termsOfServiceStatus: {
+        userId: '1234',
+        userCurrentTermsOfServiceState: TermsOfServiceState.MUST_AGREE_NOW,
+        lastAgreementDate: new Date().toISOString(),
+      },
+    })
+
+    const hook = renderHook(() => useMaybeRedirectToSignTermsOfService())
+    await waitFor(() => {
+      expect(hook.result.current.mayPromptTermsOfUse).toBe(true)
+      expect(mockNavigate).toHaveBeenCalledWith(
+        '/authenticated/signUpdatedTermsOfUse',
+      )
+    })
+  })
```

Comment:
This case is closest to the scenario reported by the user. The assertion `expect(hook.result.current.mayPromptTermsOfUse).toBe(true)` fails with the old logic and passes with the new logic.